cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -50,15 +57,36 @@ FileObject<LogCollection> LogFileManager::readBatch(
     Aws::String aws_line(line.c_str());
     Aws::Utils::Json::JsonValue value(aws_line);
     Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
-    actual_batch_size++;
-    log_set.insert(input_event);
-    data_tokens.push_back(data_token);
+    pq.push(std::make_tuple(input_event.GetTimestamp(), line, data_token));
+    if(input_event.GetTimestamp() > latestTime){
+      latestTime = input_event.GetTimestamp();
+    }
   }
+  
+  std::set<LogType, decltype(log_comparison)> log_set(log_comparison);
+  std::list<FileManagement::DataToken> data_tokens;
+  size_t actual_batch_size = 0;
+  while(!pq.empty()){
+    long curTime = std::get<0>(pq.top());
+    std::string line = std::get<1>(pq.top());
+    FileManagement::DataToken new_data_token = std::get<2>(pq.top());
+    if(latestTime - curTime < 86400000){
+      Aws::String aws_line(line.c_str());
+      Aws::Utils::Json::JsonValue value(aws_line);
+      Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
+      actual_batch_size++;
+      log_set.insert(input_event);
+      data_tokens.push_back(new_data_token);
+    }
+    pq.pop();
```

Comment:
Worth logging the case that the actual batch size is smaller given the timestamp difference?